src/auxiliary/mpi_arrays.jl

```diff
@@ -0,0 +1,332 @@
+Base.size(u::TrixiMPIArray) = size(parent(u))
+Base.getindex(u::TrixiMPIArray, idx) = getindex(parent(u), idx)
+Base.setindex!(u::TrixiMPIArray, v, idx) = setindex!(parent(u), v, idx)
+Base.IndexStyle(::Type{TrixiMPIArray{T, N, Parent}}) where {T, N, Parent} = IndexStyle(Parent)
+Base.similar(u::TrixiMPIArray, ::Type{S}, dims::NTuple{N, Int}) where {S, N}	= TrixiMPIArray(similar(parent(u), S, dims))
+Base.axes(u::TrixiMPIArray)	= axes(parent(u))
+
+
+# Implementation of the strided array interface of Base
+# See https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-strided-arrays
+Base.strides(u::TrixiMPIArray) = strides(parent(u))
+Base.unsafe_convert(::Type{Ptr{T}}, u::TrixiMPIArray{T}) where {T} = Base.unsafe_convert(Ptr{T}, parent(u))
+Base.elsize(::Type{TrixiMPIArray{T, N, Parent}}) where {T, N, Parent} = elsize(Parent)
+
+
+# TODO: MPI. Do we need customized broadcasting? What about FastBroadcast.jl and
+#            threaded execution with `@.. thread=true`?
+# See https://docs.julialang.org/en/v1/manual/interfaces/#man-interfaces-broadcasting
+
+
+# TODO: MPI. How shall we handle specializations such as `split_form_kernel!(_du::PtrArray, u_cons::PtrArray,`
+#            for `flux_ranocha_turbo` and `flux_shima_etal_turbo`?
+
+
+# Implementation of methods from ArrayInterface.jl for use with
+# LoopVectorization.jl etc.
+# See https://juliaarrays.github.io/ArrayInterface.jl/stable/
+ArrayInterface.parent_type(::Type{TrixiMPIArray{T, N, Parent}}) where {T, N, Parent} = Parent
+
+
+# TODO: MPI. Do we need LinearAlgebra methods such as `norm` or `dot`?
+
+
+# `mapreduce` functionality from Base using global reductions via MPI communication
+function Base.mapreduce(f::F, op::Op, u::TrixiMPIArray; kwargs...) where {F, Op}
+  local_value = mapreduce(f, op, parent(u); kwargs...)
+  return MPI.Allreduce(local_value, op, mpi_comm(u))
+end
+
+
+# TODO: MPI. Default settings of OrdinaryDiffEq etc.
+# Interesting options could be
+# - ODE_DEFAULT_UNSTABLE_CHECK
+# - ODE_DEFAULT_ISOUTOFDOMAIN
+# - ODE_DEFAULT_NORM
+# See https://github.com/SciML/DiffEqBase.jl/blob/master/src/common_defaults.jl
+
+# TODO: MPI. How shall we handle `length`? We want `TrixiMPIArray`s to behave
+#            like regular `Array`s in most code, e.g., for `eachindex` etc.
+#            However, we need to divide by the `length` of the global array
+#            for `ODE_DEFAULT_NORM`. We could specialize `ODE_DEFAULT_NORM`
+#            accordingly, but that requires depending on DiffEqBase (instead of
+#            SciMLBase). Alternatively, we could implement this via Requires.jl,
+#            but that will prevent precompilation and maybe trigger invalidations.
+#            Alternatively, we could specialize `length` to return a global
+#            length and make sure that all local behavior is still working as
+#            expected (if we use `eachindex` instead of `1:length` etc.).
```

Comment:
The first approach (`length` returns the global length) is the easiest to "make it work", right? Thus maybe we should start with that. Requires.jl seems the most finicky solution, while making `DiffEqBase` a dependency seems overkill if we ever want to consider making this a separate package.

Reply:
Yeah, that's my current thought, too.